pkg/hiveregistrator/registrator.go

```diff
@@ -0,0 +1,138 @@
+	InfraID                     string `json:"infraId,omitempty"`
+	ClusterID                   string `json:"clusterID,omitempty"`
+	KubeConfig                  string `json:"kubeconfig,omitempty"`
+	PullSecret                  string `json:"pullsecret,omitempty"`
+	ServicePrincipal            string `json:"serviceprincipal,omitempty"`
+
+	HiveNamespaceName         string `json:"hiveNamespaceName,omitempty"`
+	HiveClusterdeploymentName string `json:"hiveClusterDeploymentName,omitempty"`
+}
+
+type hiveRegistrator struct {
+	WorkloadCluster *WorkloadCluster
+	HiveClients     *HiveClients
+}
+
+func New(workloadcluster *WorkloadCluster, hiveclients *HiveClients) Registrator {
+	return &hiveRegistrator{
+		WorkloadCluster: workloadcluster,
+		HiveClients:     hiveclients,
+	}
+}
+
+func (hr *hiveRegistrator) Register(ctx context.Context) error {
+	kubesecretName := "admin-kube-secret"
+	pullsecretname := "pull-secret"
+	spsecretname := "serviceprincipal-secret"
+
+	namespaceName := hr.WorkloadCluster.HiveNamespaceName
+	clusterDeploymentName := hr.WorkloadCluster.HiveClusterdeploymentName
+
+	csn := ClusterNamespace(*hr.WorkloadCluster, namespaceName)
+	_, err := hr.HiveClients.Kubernetescli.CoreV1().Namespaces().Create(ctx, csn, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	kubesecret := KubeAdminSecret(*hr.WorkloadCluster, kubesecretName, namespaceName)
+
+	_, err = hr.HiveClients.Kubernetescli.CoreV1().Secrets(namespaceName).Create(ctx, kubesecret, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	pullsecret := PullSecret(*hr.WorkloadCluster, pullsecretname, namespaceName)
+	_, err = hr.HiveClients.Kubernetescli.CoreV1().Secrets(namespaceName).Create(ctx, pullsecret, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	spsecret := ServicePrincipalSecret(*hr.WorkloadCluster, spsecretname, namespaceName)
+	_, err = hr.HiveClients.Kubernetescli.CoreV1().Secrets(namespaceName).Create(ctx, spsecret, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	cds := ClusterDeployment(*hr.WorkloadCluster, kubesecretName, spsecretname, pullsecretname, namespaceName, clusterDeploymentName)
+	_, err = hr.HiveClients.Hiveclientset.HiveV1().ClusterDeployments(namespaceName).Create(ctx, cds, metav1.CreateOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (hr *hiveRegistrator) IsConnected(ctx context.Context) (bool, string, error) {
+	cd, err := hr.HiveClients.ClusterDeployment(ctx, hr.WorkloadCluster.HiveNamespaceName, hr.WorkloadCluster.HiveClusterdeploymentName)
+	if err != nil {
+		return false, "", err
+	}
+
+	for _, condition := range cd.Status.Conditions {
+		if condition.Type == v1.UnreachableCondition {
+			//checking for false, meaning not unreachable, so is reachable
+			isReachable := condition.Status != corev1.ConditionTrue
+			return isReachable, condition.Message, nil
+		}
+	}
+
+	return true, "", nil
```

Comment:
Bumping [this question](https://github.com/Azure/ARO-RP/pull/2145#discussion_r885738324):

> If `UnreachableCondition` is not present. What does it mean? Is it reachable?